odoo/addons/base/models/res_currency.py

```diff
@@ -248,26 +275,43 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        self.env['res.company'].browse({
+            vals['company_id'] for vals in vals_list if 'company_id' in vals
+        })._normalize_rate()
```

Comment:
```suggestion
        res.company_id._normalize_rate()
```

Then there is no need for the weird  `self.filtered(lambda x: x.id)`